### PR TITLE
Exercise_0

### PR DESCRIPTION
**Pull request by Arthur264**

### CODE REVIEW

File: exercise_0/main.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#define STOP_LETTER "q"
```
Comment (reczepio) [inline exercise_0/main.cpp]: Rather constexpr

File: exercise_0/main.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+struct Sounds {
+    int vowels = 0;
+    int consonants = 0;
+    int other = 0;
+};
+
+inline bool start_with_vowels(const char letter) {
+    char lower_letter = tolower(letter);
+    for (int i = 0; i < VOWELS[i]; ++i) {
```
Comment (reczepio) [inline exercise_0/main.cpp]: Please rather use standard algorithms like std::find.

File: exercise_0/main.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            ++sound.vowels;
+        } else {
+            ++sound.consonants;
+        }
+    } else {
+        ++sound.other;
+    }
+}
+
+void read_from_file(Sounds& sound, const char file_path[150]) {
```
Comment (reczepio) [inline exercise_0/main.cpp]: std::string instead of plain char array

File: exercise_0/main.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    string line;
+    ifstream infile(file_path);
+
+    if (infile.fail()) {
+        cerr << "File not exist" << endl;
+        exit(1);
+    }
+
+    while (getline(infile, line)) {
```
Comment (reczepio) [inline exercise_0/main.cpp]: This reads whole line e.g.:
"This is whole line which will be read."
You app should check each word separately.  Please use "<<" and ">>".

File: exercise_0/main.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        increment(line[0], sound);
+    }
+}
+
+void read_from_console(Sounds& sound) {
+    string word;
+    cout << "Type words (enter q for stop.)!" << std::endl;
```
Comment (reczepio) [inline exercise_0/main.cpp]: It's good idea to define constant "STOP_LETTER" but be consistent - use it everywhere.

File: exercise_0/main.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    while (cin >> word) {
+        if (word == STOP_LETTER) {
```
Comment (reczepio) [inline exercise_0/main.cpp]: Yoda style better is.

**Comment by Arthur264** [comment]
Hi 
Yes, I know it.
But I use it to compare with input word from the console. 
` while ((cin >> word) && (STOP_LETTER != word))`
And in your case, I can't compare "char" and "string" types.

What is the best way to solve this?

**Comment by reczepio** [comment]
> Hi
> Yes, I know it.
> But I use it to compare with input word from the console.
> ` while ((cin >> word) && (STOP_LETTER != word))`
> And in your case, I can't compare "char" and "string" types.
> 
> What is the best way to solve this?

E.g:
STOP_LETTER != word[0]

File: exercise_0/main.cpp [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    char lower_letter = tolower(letter);
+    for (int i = 0; i < VOWELS[i]; ++i) {
+        if (lower_letter == VOWELS[i]) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void increment(char letter, Sounds& sound) {
```
Comment (pstanisz) [inline exercise_0/main.cpp]: From design perspective the function name is not very good: what it increments? A letter? Maybe sound?
Probably there could be a better name.

**Comment by Arthur264** [comment]
@reczepio 
But in case when we have word started with "q" like "queen", it will stop while loop.

**Comment by reczepio** [comment]
@Arthur264 
std::string has methods like: size() and length().
Please see:
https://en.cppreference.com/w/cpp/string/basic_string/size
You can check size of "word" variable.

File: exercise_0/main.cpp (hunk 7 above)
Comment (Arthur264) [reply exercise_0/main.cpp]: I agree with you, changed it on 'increment_sound'.

**Comment by Arthur264** [comment]
@reczepio 
Okay, I fix it.
I have a question.
It is a good practice to use constexpr with "string" type?
`constexpr auto CONST_NAME = "const1"`

**Comment by reczepio** [comment]
@Arthur264 
You cannot use std::string in constexpr.
The type for CONST_NAME deduced in: 
constexpr auto CONST_NAME = "const1"
is char[], not std::string.

**Comment by Arthur264** [comment]
@reczepio 
I understand, thank you.

File: exercise_0/main.cpp [hunk 8]
```diff
@@ -18,7 +18,7 @@ inline bool is_vowel(const char letter) {
     return vowels.find(lower_letter) != string::npos;
 }
 
-void increment(const char letter, Sounds &sound) {
+void increment_sound(const char letter, Sounds &sound) {
```
Comment (pstanisz) [inline exercise_0/main.cpp]: The `letter` is passed by value, which means copy - const is just a conceptual here, meaning that you does not mean to change a `letter` copy.

**Review by pstanisz — APPROVED** [review]

File: exercise_0/main.cpp (hunk 8 above)
Comment (reczepio) [reply exercise_0/main.cpp]: In fact, this "const" have matter only in function body. Function declaration "drops" it.

Comment (Arthur264) [reply exercise_0/main.cpp]: @pstanisz
Can I use 'const' with value reference in function?
`void increment_sound(const char &letter, Sounds &sound)`
In order to call the function by passing values by reference and 'const' operator to prohibit changing the value.

Comment (reczepio) [reply exercise_0/main.cpp]: Yes, You can

**Pull request merged**